environments/se/core/src/main/java/org/jboss/weld/environment/se/Weld.java

```diff
@@ -632,7 +637,29 @@ public Weld setProperties(Map<String, Object> propertiesMap) {
     }
 
     /**
-     * Reset the synthetic bean archive (bean classes and enablement), explicitly added extensions and custom beans added via {@link #addBean()}.
+     * Register a per-deployment {@link Service} which is shared across the entire application.
+     * <p>
+     * Weld uses services to communicate with its environment, e.g. {@link org.jboss.weld.manager.api.ExecutorServices} or
+     * {@link org.jboss.weld.transaction.spi.TransactionServices}.
+     * </p>
+     * <p>
+     * Service implementation may specify their priority using {@link Priority}. Services with higher priority have precedence. Services that do not specify
+     * priority have the default priority of 4500.
+     * </p>
+     *
+     * @param service
+     * @return self
+     */
+    public Weld addService(Service service) {
```

Comment:
Well, adding an SPI service is not supposed to be a frequent operation. So I would like to add a single method only. But you're right that varargs variant is more practical. I can change the method signature.